### PR TITLE
Feature/viewbinding sdk 34 migration

### PR DESCRIPTION
**Pull request by fatihergin**

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/DecompressActivity.kt [hunk 1]
```diff
@@ -141,6 +146,11 @@ class DecompressActivity : SimpleActivity() {
                         continue
                     }
 
+                    val isVulnerableForZipPathTraversal = !File(newPath).canonicalPath.startsWith(parent)
+                    if (isVulnerableForZipPathTraversal) {
+                        continue
+                    }
```
Comment (fatihergin) [inline app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/DecompressActivity.kt:152]: this is not really a part of SDK 34 migration, but wanted to add a check for other pre 34 SDKs. 

please see the change for SDK 34 here [Android 14: Zip Path Traversal behavior changes](https://developer.android.com/about/versions/14/behavior-changes-14#zip-path-traversal
)

File: app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/MainActivity.kt [hunk 2]
```diff
@@ -38,18 +39,14 @@ import com.simplemobiletools.filemanager.pro.helpers.MAX_COLUMN_COUNT
...
-import kotlinx.android.synthetic.main.recents_fragment.*
-import kotlinx.android.synthetic.main.storage_fragment.*
 import me.grantland.widget.AutofitHelper
 import java.io.File
 
 class MainActivity : SimpleActivity() {
     private val BACK_PRESS_TIMEOUT = 5000
     private val MANAGE_STORAGE_RC = 201
     private val PICKED_PATH = "picked_path"
+    private lateinit var binding: ActivityMainBinding
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/MainActivity.kt:49]: lets follow calculators sample with lazy inflating

File: app/src/main/kotlin/com/simplemobiletools/filemanager/pro/adapters/ItemsAdapter.kt [hunk 3]
```diff
@@ -941,4 +932,148 @@ class ItemsAdapter(
...
+        companion object {
+            fun getByItemViewType(viewType: Int, isListViewType: Boolean): Binding {
+                return when (viewType) {
+                    TYPE_SECTION -> ItemSection
+                    TYPE_GRID_TYPE_DIVIDER -> ItemEmpty
+                    else -> {
+                        if (isListViewType) {
+                            ItemFileDirList
+                        } else {
+                            if (viewType == TYPE_DIR) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/filemanager/pro/adapters/ItemsAdapter.kt:946]: this with previous line can be merged to 1 line "else if "

**Comment by tibbi** [comment]
try initing activities like in Calculator by 
`    private val binding by lazy(LazyThreadSafetyMode.NONE) { ActivityMainBinding.inflate(layoutInflater) }`

**Comment by fatihergin** [comment]
> try initing activities like in Calculator by `private val binding by lazy(LazyThreadSafetyMode.NONE) { ActivityMainBinding.inflate(layoutInflater) }`

done

File: app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/SettingsActivity.kt [hunk 4]
```diff
@@ -134,34 +143,39 @@ class SettingsActivity : SimpleActivity() {
...
+        config.showHidden = binding.settingsShowHidden.isChecked
     }
 
     private fun setupEnablePullToRefresh() {
-        settings_enable_pull_to_refresh.isChecked = config.enablePullToRefresh
-        settings_enable_pull_to_refresh_holder.setOnClickListener {
-            settings_enable_pull_to_refresh.toggle()
-            config.enablePullToRefresh = settings_enable_pull_to_refresh.isChecked
+        binding.apply {
+            settingsEnablePullToRefresh
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/SettingsActivity.kt:152]: hm?

Comment (fatihergin) [reply app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/SettingsActivity.kt:152]: didn't get the question @tibbi

Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/SettingsActivity.kt:152]: what is that line for? Seems like some error.

Comment (fatihergin) [reply app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/SettingsActivity.kt]: ops cleanup mistake. it has no effect but yeah, i forgot to remove it.  fixed.

**Comment by tibbi** [comment]
alright thanks, it seems to work just fine at my quick test

**Pull request merged**